pkg/pipeline/encode/encode_prom.go

```diff
@@ -260,9 +260,9 @@ func NewEncodeProm(opMetrics *operational.Metrics, params config.StageParam) (En
 		cfg = *params.Encode.Prom
 	}
 
-	expiryTime := time.Duration(cfg.ExpiryTime) * time.Second
-	if expiryTime == 0 {
-		expiryTime = defaultExpiryTime
+	expiryTime := cfg.ExpiryTime
+	if expiryTime.Seconds() == 0 && expiryTime.Microseconds() == 0 {
```

Comment:
Is the following cleaner?
```suggestion
	if expiryTime.Duration == 0 {
```

Reply:
changed